datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -673,18 +673,72 @@ where
     )?))
 }
 
+/// If from_front is true, it is array_pop_front, otherwise array_pop_back
+fn general_pop_list<O: OffsetSizeTrait>(
+    array: &GenericListArray<O>,
+    from_front: bool,
+) -> Result<ArrayRef>
+where
+    i64: TryInto<O>,
+{
+    let (from_array, to_array) = if from_front {
```

Comment:
Is it better we just split this to two function? Since they are so different, without checking true for pop front or pop back may also improve readability?

Reply:
Yes, it is better for readability and maintenance.